### DISCUSSION TITLE
is '/opt/homebrew’ for M1 Macs a short-term or long-term proposition?

### DISCUSSION THREAD

**Discussion by tjluoma** [Homebrew/discussions#749] [Everyday usage]
I apologize if this is a "stupid question"… I've tried to search for an answer to this, and if it exists, I have not found it. I'm also not looking to criticize, only to understand.

As long as I can remember, `brew` users were warned to (I'm paraphrasing here): “Just use /usr/local. Anything else will probably not work.” 

Now for M1 Macs it is being suggested that we use `/opt/homebrew` although it appears that some things may still end up in `/usr/local/`.

So… is `/opt/homebrew` now the default/preferred location, or is `/opt/homebrew` the current location until `brew` for M1 Macs fully catches up to `brew` for Intel Macs?

Is that idea now that `/usr/local/` will continue to be used only for Intel installs, with the expectation that, over time, Intel installs will eventually fade away, and everyone will be using `/opt/homebrew` in the future?

Or is the idea that `/opt/homebrew` will be used until `brew` on M1 Macs can move back to `/usr/local`?

Thanks for your time.

**Comment by carlocab** [comment]
>Is that idea now that `/usr/local/` will continue to be used only for Intel installs, with the expectation that, over time, Intel installs will eventually fade away, and everyone will be using `/opt/homebrew` in the future?

Yes, this.

Note that the [docs](https://docs.brew.sh/Installation#untar-anywhere) now say

>However do yourself a favour and install to `/usr/local` on macOS Intel, `/opt/homebrew` on macOS ARM, and `/home/linuxbrew/.linuxbrew` on Linux.

**Comment by sroby** [comment]
There are known issues with Homebrew using /usr/local (most notably, other things might install stuff in that directory structure, leading to conflicts). In fact, a lot of people use competing solutions (MacPorts, etc.) specifically because they hate Brew's use of /usr/local.

A big reason why Homebrew couldn't move the default location is because bottles expect to be always installed at the same location. But since Apple Silicon is always a new installation with all new bottles, that change was possible now.

Moving away from /usr/local is a good thing, imo. I'm glad the Homebrew devs took the occasion to move it during this architecture change.

**Comment by mhumeSF** [comment]
A little confused. Is there any harm in having intel installed at `/opt/homebrew` on a new machine? Would be nice to always use one location across machines.

**Reply by SMillerDev** [reply]
Homebrew will flat out refuse to work because knowing what architecture it can expect is important for brew to function properly.
